### ISSUE TITLE
Gradle 6.6-rc-1 --configuration-cache > registration of listener on 'Gradle.buildFinished' is unsupported

### ISSUE THREAD

**Issue by douph1** [ajoberstar/grgit#322]
Trying the new gradle feature `--configuration-cache` I have this error

Plugin version : 4.0.2

```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.
Gradle can be made to ignore these problems, see https://docs.gradle.org/6.6-rc-1/userguide/configuration_cache.html#config_cache:usage:ignore_problems.

1 problem was found storing the configuration cache.
- plugin 'org.ajoberstar.grgit': registration of listener on 'Gradle.buildFinished' is unsupported
  See https://docs.gradle.org/6.6-rc-1/userguide/configuration_cache.html#config_cache:requirements:build_listeners

See the complete report at file:///.../build/reports/configuration-cache/dg0qms2iixa405dr5cw20kfzz-2/configuration-cache-report.html
> Listener registration 'Gradle.buildFinished' by build 'xxx' is unsupported.
```

**Comment by rusefillc** [comment]
+1 :)

**Comment by kibotu** [comment]
+1

**Comment by Petrakeas** [comment]
+1

**Comment by yongfa365** [comment]
\.gradle\gradle.properties:

org.gradle.caching=false
org.gradle.unsafe.configuration-cache=false

**Comment by kibotu** [comment]
the issue is about supporting gradle configuration cache not how to disabled it @yongfa365

**Comment by abelom** [comment]
by the way #346 mentions configuration change in connection with next major version

**Comment by ajoberstar** [comment]
Support will be available as of 5.0.0-rc.1 (working on releasing right now). I'm going to try adopting this new approach in gradle-git-publish and reckon before calling 5.0.0 final. Please provide feedback on the behavior in new issues if you're looking for further changes.
